extensions/positron-r/src/kernel-spec.ts

```diff
@@ -149,10 +149,17 @@ export function createJupyterKernelSpec(
 		'kernel_protocol_version': '5.5' // eslint-disable-line
 	};
 
-	// Unless the user has chosen to restore the workspace, pass the
-	// `--no-restore-data` flag to R.
-	if (!config.get<boolean>('restoreWorkspace')) {
-		kernelSpec.argv.push('--no-restore-data');
+	// For temporary, approximate backward compatibility, check both
+	// 'saveAndRestoreWorkspace' and 'restoreWorkspace', which was deprecated in
+	// late October 2025. Remove the latter setting and check in a future release.
+	const shouldSaveAndRestore = config.get<boolean>('saveAndRestoreWorkspace') || config.get<boolean>('restoreWorkspace');
+
+	if (shouldSaveAndRestore) {
+		// '--restore-data' is the default but let's be explicit for clarity and
+		// symmetry with the other branch
+		kernelSpec.argv.push('--restore-data', '--save');
+	} else {
+		kernelSpec.argv.push('--no-restore-data', '--no-save');
```

Comment:
I think we should go straight to the desirable state of sending either

```
--no-restore-data --no-save
```

or

```
--restore-data --save
```

I can't really imagine who or why anyone was using the old config that would restore `.RData` but never write to it. So I propose we go straight to the end game.